### PR TITLE
refactor: mention criteria to include hyphens

### PR DESCRIPTION
**Pull request by sshanzel**
Allow usage of hyphens when mentioning a user.

### CODE REVIEW

File: __tests__/comments.ts [hunk 1]
```diff
@@ -330,7 +330,10 @@ describe('query commentPreview', () => {
     expect(res.errors).toBeFalsy();
     expect(res.data.commentPreview).toMatchSnapshot();
 
-    const mention = '@Lee';
+    await con
+      .getRepository(User)
+      .update({ username: 'Lee' }, { username: 'Lee-' });
+    const mention = '@Lee-';
```
Comment (rebelchris) [inline __tests__/comments.ts:336]: Maybe suggest to append something at the back for safety tests

**Review by rebelchris — APPROVED** [review]
Just a small note to improve test cases

File: __tests__/comments.ts (hunk 1 above)
Comment (sshanzel) [reply __tests__/comments.ts:336]: Good point. Let me add it as a separate one.

**Review by rebelchris — APPROVED** [review]
Thanks! 🎉

**Review by idoshamun — APPROVED** [review]
Awesome!

**Comment by capJavert** [comment]
@sshanzel this will only fix new comments?

Old comments will still be broke or mention wrong person (in case of reported issue from the task)?

**Comment by sshanzel** [comment]
> Old comments will still be broken or mention wrong person (in case of reported issue from the task)?

Yes, this was not intended to fix existing mentions. I will check with the produce team if they want the old ones to be fixed.

**Pull request merged**